### DISCUSSION TITLE
Is there a way to generate a dll or lib that can then be used with c/c++?

### DISCUSSION THREAD

**Discussion by bovacu** [aardappel/lobster#250] [Q&A] [answered]
Hello! I'm fairly new to the language and I saw there is a way to compile lobster to cpp, which I did and saw the cpp that was generated, tried to create a lib from it but seems i did not succeed. Is there a way or is it in the plans to make something like this? 

Thanks!

**Comment by aardappel** [comment, marked as answer]
It's hard to help you when you say "i did not succeed" instead of detailed errors and a description of what you did.

Please make sure you follow the instructions here https://aardappel.github.io/lobster/implementation.html on how to build the `--cpp` output.

This typically generates an executable. There may be a way to generate it into a dll, but even so it would not generate C/C++ compatible function symbols for all Lobster functions, if that's what you were expecting. Lobster is not linked-compatible with C/C++ functions, it is a higher level language that needs explicit bindings, see the same file above.

**Reply by bovacu** [reply]
Oh yeah sorry,

`There may be a way to generate it into a dll, but even so it would not generate C/C++ compatible function symbols for all Lobster functions, if that's what you were expecting`

That was my problem in fact I manage to create the .lib and when linking it could not find the symbols inside, so probably it wasn't built correctly and then I tried to create the dll (same way as the .lib with Visual Studio) and it gave about 150 errors. I was just wondering if there was a direct way but as you said, there is not currently, on top of maybe not all functions being compatible wih C++.

Thanks!
